presto-native-execution/presto_cpp/main/PrestoMain.cpp

```diff
@@ -37,3 +38,12 @@ folly::Singleton<facebook::velox::BaseStatsReporter> reporter([]() {
   return new facebook::velox::DummyStatsReporter();
 });
 #endif
+
+#ifndef PRESTO_MEMORY_CHECKER_TYPE
```

Comment:
This block is not needed.

Reply:
@majetideepak We need this code block here to serve as a dummy checker like how DummyStatsReporter is created above this code block. I tried running the presto server on a machine that's not Linux and it errored out because it was looking for a singleton of type PeriodicMemoryChecker. This code block here would create a dummy PeriodicMemoryChecker singleton for whenever a PRESTO_MEMORY_CHECKER_TYPE is not defined/specified.

Reply:
try returning a nullptr and see. We can then remove the Dummy implementation.
```
folly::Singleton<facebook::presto::PeriodicMemoryChecker> checker([]() {
  return nullptr;
});
```

Reply:
This worked, removing Dummy class implementation.

Reply:
Can we move this from PrestoMain.cpp to the PeriodicMemoryChecker.cpp. We don't need to pollute this with the singleton here.